src/Http/ServerRequest.php

```diff
@@ -625,6 +635,11 @@ public function __call($name, $params)
      */
     public function __get($name)
     {
+        deprecationWarning(
+            'Accessing routing parameters through __get will removed in 4.0.0. ' .
```

Comment:
A message like `"Accessing '{$name}' as property has been deprecated and will be removed in 4.0.0"` might be more easily understandable to novice users.